EIPS/eip-1962.md

```diff
@@ -0,0 +1,224 @@
+    - non-residue for Fp6 extension (that is "3 over 2"). This is an element in Fp2 and encoded using the rules from the beginning of this EIP.
+- Encoding of the "A" parameter in Fp2 for the curve over the extension (Fp2).
+- Encoding of the "B" parameter in Fp2 for the curve over the extension (Fp2).
+
+For BLS12 family `x` parameter is enough for gas costs estimation. It's Hamming weight and bit length determines complexity of the Miller loop. Final exponentiation depends of the base field size.
+
+Proposed gas cost formula: `C_0 * quadratic(fe_length) + N * C_1 * fe_length * Hamming_and_bit_length(x)`
+
+#### Pairing operation interface for other families
+
+Complexity for BN family is also determined by the single scalar parameter, so interface will be similar to BLS12. For MNT4/6 and Cocks-Pinch curves there would be required to specify number of steps in the Miller loop explicitly in the interface.
+
+### Possible simplifications
+
+Due to high complexity of the proposed operations in the aspects of implementation, debugging and evaluation of the factors for gas costs it may be appropriate to either limit the set of curves at the moment of acceptance to some list and then extend it. Another approach (if it's technically possible) would be to have the "whilelist" contract that can be updated without consensus changes (w/o fork).
+
+In the case of limited set of curve the following set is proposed as a minimal:
+- BN254 curve from the current version of Ethereum
+- BN curve from DIZK with 2^32 roots of unity
+- BLS12-381 
+- BLS12-377 from ZEXE with large number of roots of unity
+- MNT4/6 cycle from the original [paper](https://eprint.iacr.org/2014/595.pdf). It's not too secure, but may give some freedom for experiments.
+- MNT4/6 cycle from Coda if performance allows
+- Set of CP generated curves that would allow embedding of BLS12-377 and may be some BN curve that would have large power of two divisor for both base field and scalar field modulus (example of CP curve for BLS12-377 can be found in ZEXE). 
+
+## Rationale
+
+Only the largest design decisions will be covered:
+- While there is no arithmetic over the scalar field (which is modulo size of the main group) of the curve, it's required for gas estimation purposes.
+- One may separate interfaces for additions, multiplications and multiexponentiations and estimate gas costs differently for every operation, but it would bring confusion for users and will make it harder to use a precompile from the smart-contract.
+- Multiexponentiation is a separate operation due to large cost saving
+- There are no point decompressions due to impossibility to get universal gas estimation of square root operation. For a limited number of "good" cases prices would be too different, so specifying the "worst case" is expensive and inefficient, while introduction of another level if complexity into already complicated gas costs formula is not worth is.
+
+## EIP 1109
```

Comment:
Is this part of the rationale? Can you move this into a subheading within rationale?

Reply:
Moved to subchapter